qemu/tests/blockdev_commit_with_ignore.py

```diff
@@ -0,0 +1,80 @@
+from avocado.utils import process
+
+from virttest import utils_misc
+
+from provider import backup_utils
+from provider import job_utils
+from provider.blockdev_commit_base import BlockDevCommitTest
+
+
+class BlockdevCommitWithIgnore(BlockDevCommitTest):
+
+    def generate_tempfile(self, root_dir, filename="data",
+                          size="1000M", timeout=360):
+        backup_utils.generate_tempfile(
+            self.main_vm, root_dir, filename, size, timeout)
+        self.files_info.append([root_dir, filename])
+
+    def wait_until_block_job_error(self, job_id, timeout):
+        """Block until block job reach error"""
+        def _wait_until_block_job_error():
+            checked = False
+            for event in self.main_vm.monitor.get_events():
+                if event.get("event") != "BLOCK_JOB_ERROR":
+                    continue
+                data = event.get("data", dict())
+                if job_id == data["device"] and data["action"] == "ignore":
+                    checked = True
+                    break
+            return checked
+
+        checked = utils_misc.wait_for(
+            _wait_until_block_job_error,
+            first=0.1,
+            timeout=timeout)
+        return checked
+
+    def commit_snapshots(self):
+        device = self.params["device_tag"].split()[0]
+        device_params = self.params.object_params(device)
+        snapshot_tags = device_params["snapshot_tags"].split()
+        self.device_node = self.get_node_name(device)
+        device = self.get_node_name(snapshot_tags[-1])
+        arguments = {}
+        arguments.update({"on-error": "ignore"})
+        cmd, arguments = backup_utils.block_commit_qmp_cmd(device, **arguments)
+        timeout = self.params.get("job_timeout", 600)
+        self.main_vm.monitor.cmd(cmd, arguments)
+        job_id = arguments.get("job-id", device)
+        job_error = self.wait_until_block_job_error(job_id, timeout)
```

Comment:
We may use:
job_error = get_event_by_condition(self.main_vm, 'BLOCK_JOB_ERROR', timeout, device=job_id)